### PR TITLE
Add support to Talon SRX for an integrated absolute encoder

### PR DESCRIPTION
**Pull request by WispySparks**
First time working around the codebase, I went this route because I don't think you can really have a reference to a talon's sensor like you can with revlib. Not sure if this all looks right

### CODE REVIEW

File: src/main/java/swervelib/parser/json/DeviceJson.java [hunk 1]
```diff
@@ -188,7 +190,9 @@ public SwerveMotor createMotor(boolean isDriveMotor)
       case "talonfx":
         return new TalonFXSwerve(id, canbus != null ? canbus : "", isDriveMotor);
       case "talonsrx":
-        return new TalonSRXSwerve(id, isDriveMotor);
+        return new TalonSRXSwerve(id, isDriveMotor, FeedbackDevice.QuadEncoder);
+      case "talonsrx_pwm":
```
[the remark this answers is not in the document]
Comment (WispySparks) [reply src/main/java/swervelib/parser/json/DeviceJson.java]: is this an alright name for the encoder section?

**Comment by WispySparks** [comment]
Okay it might be a while until this is able to be tested lol but I'll let you know

**Comment by WispySparks** [comment]
Our swerve is in the process of being built over the summer, we still need to make a belly plate and wire stuff up but we only meet once a week. I'll let you know once we're testing stuff

**Comment by WispySparks** [comment]
Still working through haha, at the point of tuning PID now I believe

**Comment by WispySparks** [comment]
Currently we're doing a bandaid fix for selecting the current sensor into the TalonSRX because for some reason it's getting set to QuadEncoder instead of PulseWidth. I've looked through my pr'ed code and YAGSL code for a while but still haven't found the issue so we're doing this for now to get swerve up and running.
```java
swerveDrive = parser.createSwerveDrive(SwerveK.maxRobotSpeed.in(MetersPerSecond), angleConversionFactor, driveConversionFactor);
for (var mod : swerveDrive.getModules()) { // Bandaid Solution
    var motor = (WPI_TalonSRX) mod.getAngleMotor().getMotor();
    motor.configSelectedFeedbackSensor(FeedbackDevice.PulseWidthEncodedPosition);
}
```

**Comment by cmdada** [comment]
If you need a swerve bot with talons to test on I have access to am swerve with talon srxs on turning and would happily test code for it

**Comment by WispySparks** [comment]
I'll be in the shop tomorrow testing that last change for the sensor being selected (I only did it in sim so far) and then this should be good to go.

**Comment by WispySparks** [comment]
Everything looks good on my end, this should be good to merge.

**Comment by thenetworkgrinch** [comment]
Almost i need to merge `beta` into `dev` then add a new function i recently added to the `SwerveMotor.java`

**Comment by thenetworkgrinch** [comment]
Could you resolve the conflicts when you get a chance?

**Comment by thenetworkgrinch** [comment]
Just in time for the 2025 release!

**Pull request merged**